fem/ceed/solvers/full-assembly.cpp

```diff
@@ -81,12 +81,19 @@ int CeedSingleOperatorFullAssemble(CeedOperator op, SparseMatrix *out)
       ierr = CeedOperatorFieldGetVector(input_fields[i], &vec); PCeedChk(ierr);
       if (vec == CEED_VECTOR_ACTIVE)
       {
-         ierr = CeedOperatorFieldGetBasis(input_fields[i], &basisin);
-         PCeedChk(ierr);
+         if (!basisin)
+         {
+            ierr = CeedOperatorFieldGetBasis(input_fields[i], &basisin);
+            PCeedChk(ierr);
+         }
          ierr = CeedBasisGetNumComponents(basisin, &ncomp); PCeedChk(ierr);
          ierr = CeedBasisGetDimension(basisin, &dim); PCeedChk(ierr);
-         ierr = CeedOperatorFieldGetElemRestriction(input_fields[i], &rstrin);
-         PCeedChk(ierr);
+         ierr = CeedBasisDestroy(&basisin); PCeedChk(ierr);
```

Comment:
@jeremylt Why do we need the line 84 conditional for non-NULL `basisin` if it always gets destroyed here on line 91? And what does that mean down on line 181 when the (now-NULL) `basisin` gets used (I think formerly assuming it was the last field)? This code must not be tested properly, which makes me apprehensive about making changes here.

Reply:
Oh, this is using an old pattern that we removed from libCEED previously. I'll update this (and figure out how I misunderstood the test suite as apparently I ran it but not the libCEED parts?)